File: aon-web-aio/src/main/webapp/models/ExemptType.js
export const EXEMPT_TYPES = Object.freeze({
    EMPTY: {
        value: 'EMPTY',
        name: '------------'
    },
    NO_SOFTWARE: {
        value: 'NO_SOFTWARE',
        name: 'No utiliza sistema informático de facturación'
    },
    NO_OBLIGATION: {
        value: 'NO_OBLIGATION',
        name: 'Operaciones sin obligación de emitir factura'
    }, 
    REAGP: {
        value: 'REAGP',
        name: 'REAGP sin emisión de factura propia'
    },
    AUTHORIZATION: {
        value: 'AUTHORIZATION',
        name: 'Exenci\u00F3n autorizada por la administración tributaria'
    },
});

export class ExemptType {
    value;
    name;

    constructor(value) { 
        Object.keys(EXEMPT_TYPES).forEach(key => {
            if (EXEMPT_TYPES[key].value === value) {
                this.value = EXEMPT_TYPES[key].value;
                this.name = EXEMPT_TYPES[key].name;
            }
        });
    }
   
}

File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/type/ExemptType.java
package com.esferalia.aon.occam.api.model.type;

import java.util.Optional;

import com.esferalia.aon.watson.util.AonCollectionUtils;
import com.esferalia.aon.watson.util.AonStringUtils;

public enum ExemptType {
	NO_SOFTWARE		("No utiliza sistema inform\u00E1tico de facturaci\u00F3n"),
	NO_OBLIGATION	("Operaciones sin obligaci\u00F3n de emitir factura"),
    REAGP			("REAGP sin emisi\u00F3n de factura propia"),
    AUTHORIZATION	("Exenci\u00F3n autorizada"),
    ;

    private final String description;

    private ExemptType(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }

	public static Optional<ExemptType> safeValueOf(String cause) {
		if (AonStringUtils.isBlank(cause)) return Optional.empty();
		String c = AonStringUtils.trimToNull(cause);
		return AonCollectionUtils.stream(values())
			.filter(et -> AonStringUtils.equalsIgnoreCase(et.name(), c))
			.findFirst();
	}
}
